src/main/java/service/operation/PurchaseOperation.java

```diff
@@ -0,0 +1,22 @@
+package service.operation;
+
+import dao.FruitDao;
+import dao.FruitDaoImpl;
+
+public class PurchaseOperation implements OperationHandler {
+    private FruitDao fruitDao;
+
+    public PurchaseOperation() {
+        this.fruitDao = new FruitDaoImpl();
```

Comment:
Following the SOLID principles, the `FruitDao` should not be instantiated directly in the `PurchaseOperation` class. Instead, it should be passed as a dependency through the constructor, allowing for better flexibility and easier testing. You can make the `fruitDao` final to ensure it's not reassigned after construction.